docker/setup/terraform/02_data_rg.tf

```diff
@@ -0,0 +1,280 @@
+# ------------------------------
+# Data Resource Group
+# ------------------------------
+
+# Create a data resource group if it doesn't exist
+resource "azurerm_resource_group" "data" {
+  name     = "${var.RESOURCE_GROUP_NAME}data"
+  location = var.location
+
+  tags = {}
+}
+
+# Generate random text for a unique storage account name
+resource "random_id" "randomId" {
+  keepers = {
+    # Generate a new ID only when a new resource group is defined
+    resource_group = azurerm_resource_group.vm.name
+  }
+  
+  byte_length = 3
+}
+
+# Create server table storage account if it doesn't exist
+resource "azurerm_storage_account" "serverTablesName" {
+  name                      = "${var.serverTablesName}${random_id.randomId.hex}"
+  resource_group_name       = azurerm_resource_group.data.name
+  location                  = azurerm_resource_group.data.location
+  account_kind              = "Storage"
+  account_tier              = "Standard"
+  account_replication_type  = "GRS"
+  allow_blob_public_access  = true

```

Comment:
If I recall correctly, this storage account is only ever used inside of the cloud services, so I believe we can turn off public access.

Reply:
Similar comment for resources below.